File: src/main/java/me/miki/shindo/utils/file/filter/PngFileFilter.java
package me.miki.shindo.utils.file.filter;

import me.miki.shindo.utils.file.FileUtils;

import javax.swing.filechooser.FileFilter;
import java.io.File;

public class PngFileFilter extends FileFilter {

    @Override
    public boolean accept(File file) {

        if (file.isDirectory()) {
            return true;
        }

        String extension = FileUtils.getExtension(file);

        return extension != null && extension.equalsIgnoreCase("png");
    }

    @Override
    public String getDescription() {
        return "Png Images (*.png)";
    }
}

File: src/main/kotlin/me/miki/shindo/utils/file/filter/image/PngFileFilter.kt
package me.miki.shindo.utils.file.filter.image

import me.miki.shindo.utils.file.FileUtils
import java.io.File
import javax.swing.filechooser.FileFilter

class PngFileFilter : FileFilter() {
    override fun accept(file: File?): Boolean {
        if (file == null) return false
        if (file.isDirectory) return true
        val extension = FileUtils.getExtension(file)
        return extension != null && extension.equals("png", ignoreCase = true)
    }

    override fun getDescription(): String = "Png Images (*.png)"
}
